server/utils.js

```diff
@@ -0,0 +1,29 @@
+//Probably should move more stuff into here
+
+var shortid = require("shortid")
+var logger = require("./logger")
+
+var slugify = function(str) {
+	str = str.replace(/[ \t\n_]+/g, "_")
+	str = str.replace(/[^\w_]+/g, "")
+	return str.toLowerCase()
+}
+
+//Recursive promises, weeee....
+var generateId = function(db) {
+	return new Promise(function(resolve, reject) {
+		var id = shortid.generate()
+		db.find({guid: id}, function(e, data) {
```

Comment:
`findOne` might be a better choice here, but the change probably isn't necessary